packages/typespec-client-generator-core/src/decorators.ts

```diff
@@ -1419,6 +1421,40 @@ export function getLegacyHierarchyBuilding(context: TCGCContext, target: Model):
   return getScopedDecoratorData(context, legacyHierarchyBuildingKey, target);
 }
 
+const markAsLroKey = createStateSymbol("markAsLro");
+
+export const $markAsLro: MarkAsLroDecorator = (
+  context: DecoratorContext,
+  target: Operation,
+  scope?: LanguageScopes,
+) => {
+  if (target.returnType.kind !== "Model") {
+    reportDiagnostic(context.program, {
+      code: "invalid-mark-as-lro-target",
+      format: {
+        operation: target.name,
+      },
+      target: context.decoratorTarget,
+    });
+    return;
+  }
+  if (getLroMetadata(context.program, target)) {
```

Comment:
If an operation is not defined according to the LRO pattern, I doubt if `getLroMetadata` could get the correct result. It is not safe to use the result directly, especially for the final result.